master/internal/db/postgres_agent.go

```diff
@@ -17,12 +21,18 @@ WHERE NOT EXISTS (
 }
 
 // EndAgentStats updates the end time of an instance.
-func (db *PgDB) EndAgentStats(a *model.AgentStats) error {
-	return db.namedExecOne(`
-UPDATE agent_stats
-SET end_time = (SELECT CURRENT_TIMESTAMP)
-WHERE agent_id = :agent_id AND end_time IS NULL
-`, a)
+func EndAgentStats(a *model.AgentStats) error {
+	res, err := Bun().NewUpdate().Table("agent_stats").Set(
+		"end_time = (SELECT CURRENT_TIMESTAMP)").Where(
+		"agent_id = ?", a.AgentID).Where("end_time IS NULL").Exec(context.TODO())
+	if err != nil {
+		return err
+	}
+	count, err := res.RowsAffected()
```

Comment:
when zero rows is affected, is `err` here nil or not?

Reply:
`err` here will be nil no matter how many rows are affected